testing/test_rx_tx/test_rx_tx.py

```diff
@@ -0,0 +1,388 @@
+import numpy as np
+import uhd.usrp as usrp
+import uhd
+import math
+import cmath
+import sys
+import time
+import threading
+
+# Define constants
+ADDR = "num_recv_frames=512,num_send_frames=256,send_buff_size=2304000"
+CLKSRC = "internal"
+
+RXSUBDEV = "A:A A:B"
+TXSUBDEV = "A:A"
+
+TXCHAN = np.array([0])
+RXCHAN = np.array([0, 1])
+
+TXRATE = 250.0E3
+RXRATE = 250.0E3
+TXFREQ = 11E6
+RXFREQ = 11E6
+DELAY = 10E-3
+
+PULSETIMES = [0.0, 13500E-3, 18000E-6, 30000E-6, 33000E-6, 39000E-6, 40500E-6]
+
+start_tx = False
+start_time = uhd.types.TimeSpec(0)
+tx_ringbuffer_size = 10000
+test_mode = "0"
+
+# Make ramped pulses
+def make_ramped_pulse(tx_rate: float):
+	"""
+	Purpose:
+		Create a ramped pulse to be used during testing
+	Pre-conditions:
+		tx_rate: the data transfer rate of the device, double format
+	Return:
+		a list of complex numbers representing the generated pulse
+	"""
+	# define amplitude and pulse length
+	amp = 1.0/math.sqrt(2.0)
+	pulse_len = 300.0E-6
+
+	tr_start_pad = 60
+	tr_end_pad = 60
+	num_samps_per_antenna = int(np.ceil(pulse_len * (tx_rate + tr_start_pad + tr_end_pad)))
```

Comment:
I know you copied this from the c++ code, but I think these units dont work out. Maybe I'm just missing something.